src/charm.py

```diff
@@ -88,25 +88,31 @@ def __init__(self, *args):
             resource_reqs_func=self._resource_reqs_from_config,
         )
 
+        self.ingress = IngressPerUnitRequirer(self, relation_name="ingress", port=self._port)
+
         self._topology = JujuTopology.from_charm(self)
 
+        external_url = urlparse(self.external_url)
+
+        # TODO when https://github.com/canonical/traefik-k8s-operator/issues/78 is fixed, change
+        #  the refresh event to self.ingress.on.ready_for_unit.
         self._scraping = MetricsEndpointProvider(
             self,
             relation_name="self-metrics-endpoint",
             jobs=self.self_scraping_job,
+            external_hostname=external_url.netloc,
+            refresh_event=self.on.update_status,  # needed for ingress
```

Comment:
Shouldn't we set more refresh_events, like??

```python
            refresh_event=[
                self.ingress_per_unit.on.ready_for_unit,
                self.ingress_per_unit.on.revoked_for_unit,
                self.on.ingress_relation_departed,
                self.on.update_status,
            ],
```

Reply:
Fixed, thanks!